cmd/acra-keys/keys/list-keys_test.go

```diff
@@ -74,6 +83,165 @@ func TestPrintKeysJSON(t *testing.T) {
+		entryTime, err := time.Parse(filesystem.HistoricalFileNameTimeFormat, entry.Name())
+		if err != nil {
+			t.Fatal(err)
+		}
+		pubKeysTimes = append(pubKeysTimes, entryTime)
+	}
+
+	privateKeysEntries, err := os.ReadDir(filepath.Join(dirName, string(clientID)+"_storage.old"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	privateKeysTimes := make([]time.Time, 0, timesToGenerateHistoricalKeys)
+	for _, entry := range privateKeysEntries {
+		entryTime, err := time.Parse(filesystem.HistoricalFileNameTimeFormat, entry.Name())
+		if err != nil {
+			t.Fatal(err)
+		}
+		privateKeysTimes = append(privateKeysTimes, entryTime)
+	}
+
+	descriptions, err := store.ListHistoricalKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(descriptions) != 2*timesToGenerateHistoricalKeys {
+		t.Fatal("Expect exact number of historical keys description")
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if descriptions[i].CreationTime.String() != privateKeysTimes[i].String() {
+			t.Fatalf("Not expected creation time of historical private key, %s not equal %s", descriptions[i].CreationTime.String(), privateKeysTimes[0].String())
+		}
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if descriptions[i+3].CreationTime.String() != pubKeysTimes[i].String() {
+			t.Fatalf("Not expected creation time of historical public key, %s not equal %s", descriptions[i].CreationTime.String(), privateKeysTimes[0].String())
+		}
+	}
+}
+
+func TestListHistoricalKeysV2(t *testing.T) {
+	timesToGenerateHistoricalKeys := 3
+	dirName := t.TempDir()
+	if err := os.Chmod(dirName, 0700); err != nil {
+		t.Fatal(err)
+	}
+
+	clientID := []byte("testclientid")
+
+	keyloader.RegisterKeyEncryptorFabric(keyloader.KeystoreStrategyEnvMasterKey, env_loader.NewEnvKeyEncryptorFabric(keystore.AcraMasterKeyVarName))
+	masterKey, err := keystoreV2.NewSerializedMasterKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+	flagSet := flag.NewFlagSet(CmdMigrateKeys, flag.ContinueOnError)
+	keyloader.RegisterCLIParametersWithFlagSet(flagSet, "", "")
+
+	err = flagSet.Set("keystore_encryption_type", keyloader.KeystoreStrategyEnvMasterKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Setenv(keystore.AcraMasterKeyVarName, base64.StdEncoding.EncodeToString(masterKey))
+
+	listCMD := &ListKeySubcommand{
+		CommonKeyStoreParameters: CommonKeyStoreParameters{
+			keyDir: dirName,
+		},
+		CommonKeyListingParameters: CommonKeyListingParameters{
+			historicalKeys: true,
+		},
+		FlagSet: flagSet,
+	}
+
+	store, err := openKeyStoreV2(listCMD)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = store.GenerateDataEncryptionKeys(clientID); err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < timesToGenerateHistoricalKeys; i++ {
+		if err = store.GenerateDataEncryptionKeys(clientID); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	descriptions, err := store.ListHistoricalKeys()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(descriptions) != timesToGenerateHistoricalKeys {
```

Comment:
description contains public and private keys together? can we check it and validate? for example, we can remember all keys generated above with an order and then compare them. Validate that first generated is last rotated key and last generated is first key. Also, we can compare public content of keys